### DISCUSSION TITLE
How to handle loaded data outside of valid value range

### DISCUSSION THREAD

**Discussion by luukvdmeer** [ZGIS/semantique#33] [Ideas]
The layout file specifies for each layer which values it may contain. For categorical data this is a list of each possible value, for numerical data it is a lower and upper bound. Right now the retriever functions do not mask data that are outside of these valid ranges. This is mainly important for those layers that refer to the same actual data source, like "atmosphere colortype" and "appearance colortype". For example, the retrieved "appearance colortype" layer still contains category indices of the cloud category but this index does not appear in the value labels (since the metadata object of the layer does not contain it). We cannot really remove those values, because they are not the same as "missing data". How to handle them?

**Comment by whisperingpixel** [comment]
Let's have a look at the options that we have:

1. Setting it to None (removing)
2. Create a new category e.g., (valid) observation
3. Replacing the values with the overall concept (e.g., atmosphere, artefact)

1: Mixes the semantics of missing values
2: Does not really solve the problem
3: Does not really solve the problem

Any other ideas?

My preference with the given options is still (1)

**Reply by dtiede** [reply]
I cannot make an informed decision, maybe we can discuss in person.
